validator/resource_value_pattern_test.rego

```diff
@@ -0,0 +1,187 @@
+package templates.gcp.GCPResourceValuePatternConstraintV1
+
+import data.test.fixtures.resource_value_pattern.assets as fixture_assets
+import data.test.fixtures.resource_value_pattern.constraints.list as fixture_constraints
+
+# Helper to lookup a constraint based on its name via metadata, not package
+lookup_constraint[name] = [c] {
+	c := fixture_constraints[_]
+	c.metadata.name = name
+}
+
+# Helper to execute constraints against assets
+find_violations[violation] {
+	asset := data.assets[_]
+	constraint := data.test_constraints[_]
+	issues := deny with input.asset as asset with input.constraint as constraint
+	total_issues := count(issues)
+	violation := issues[_]
+}
+
+# Helper to create a set of resource names from violations
+resource_names[name] {
+	# Not sure why I need this, data.violations was a array_set but unless
```

Comment:
I'm not able to repro this issue. Maybe it's an OPA versioning thing? I'm using OPA 0.11.

alternatively, we can use a function for this:
```
found_resources(violations) = output {
	output := {x | x := violations[_].details.resource}
}
```

that seems more direct?